### PR TITLE
watcher: add version 0.11.0, remove older versions

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **watcher/0.11.0**

#### Motivation
There are several improvements in 0.11.0.

#### Details
https://github.com/e-dant/watcher/compare/release/0.10.1...release/0.11.0

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/watcher/all/conandata.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 sources:
+  "0.11.0":
+    url: "https://github.com/e-dant/watcher/archive/release/0.11.0.tar.gz"
+    sha256: "dd92496d77b6bc27e27ed28253faae4d81bc58b19407d154bb49504b2af73664"
   "0.10.1":
-    url: "https://github.com/e-dant/watcher/archive/release/0.10..tar.gz"
-    sha256: "3c2d3318f64d79437b1305a83382aa26e5054444ac4aed1821937cb97fedc03a"
+    url: "https://github.com/e-dant/watcher/archive/release/0.10.1.tar.gz"
```
Comment (AbrilRBS) [inline recipes/watcher/all/conandata.yml:6]: !!!! Nice catch! Looks like this broken link downloads the latest available version? That's extremelly unexpected!

**Review by AbrilRBS — APPROVED** [review]
Thanks!

File: recipes/watcher/all/conanfile.py [hunk 2]
```diff
@@ -19,6 +18,7 @@ class WatcherConan(ConanFile):
     topics = ("watch", "filesystem", "event", "header-only")
     package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
```
Comment (AbrilRBS) [inline recipes/watcher/all/conanfile.py:21]: <3

**Review by uilianries — APPROVED** [review]
Nice, thank you for fixing the version bug too!!

**Pull request merged**